typhos/display.py

```diff
@@ -1376,6 +1381,9 @@ def add_device(self, device, macros=None):
             self.devices.clear()
         # Add the device to the cache
         super().add_device(device)
+        logger.debug("Registering signals from device %s", device.name)
+        for component_walk in device.walk_signals():
+            register_signal(component_walk.item)
```

Comment:
Probably the biggest impact for the general case is this loop here. I imagine it's not bad performance-wise, though.

Reply:
It shouldn't matter too much, but there was probably an alternate path where we could stash the devices and have the sig:// plugin find the signal at runtime, a fast process if we use the dotted names.